SimCalorimetry/HcalTrigPrimAlgos/src/HcalTriggerPrimitiveAlgo.cc

```diff
@@ -513,9 +504,31 @@ void HcalTriggerPrimitiveAlgo::analyzeHF2016(
     
 }
 
+bool
+HcalTriggerPrimitiveAlgo::validChannel(const QIE10DataFrame& digi, int ts) const
+{
+   auto mask = conditions_->getHcalTPChannelParameter(HcalDetId(digi.id()))->getMask();
+   if (mask)
+      return false;
+
+   auto parameters = conditions_->getHcalTPParameters();
+   auto adc_threshold = parameters->getADCThresholdHF();
+   auto tdc_mask = parameters->getTDCMaskHF();
+
+   if (override_parameters_.exists("ADCThresholdHF"))
```

Comment:
Hi @matz-e - we can't do this sort of check for every digi. can you see who to move it to a once per job check. Thanks